includes/class-sensei-course.php

```diff
@@ -2995,6 +2995,41 @@ public static function the_course_enrolment_actions() {
 		<?php
 	}
 
+	/**
+	 * Check if a user can manually enrol themselves.
+	 *
+	 * @param int $course_id Course post ID.
+	 *
+	 * @return bool
+	 */
+	public static function can_current_user_manually_enrol( $course_id ) {
+		if ( ! is_user_logged_in() ) {
+			return false;
+		}
+
+		// Check if the user is already enrolled through any provider.
+		$is_user_enrolled = self::is_user_enrolled( $course_id, get_current_user_id() );
+
+		$default_can_user_manually_enrol = ! $is_user_enrolled;
+
+		$can_user_manually_enrol = apply_filters_deprecated(
+			'sensei_display_start_course_form',
+			[ $default_can_user_manually_enrol, $course_id ],
+			'3.0.0',
+			'sensei_can_user_manually_enrol'
+		);
+
+		/**
+		 * Check if currently logged in user can manually enrol themselves. Defaults to `true`.
```

Comment:
Should this say "Defaults to `true` if they are not already enrolled"?

Reply:
Yup! That was a last second change. bf499cca78737f971ea396f40ba82051c9919560